domain/model/service/service.go

```diff
@@ -590,6 +590,13 @@ func (s *Service) UpdateCredential(
 	return s.st.UpdateCredential(ctx, uuid, key)
 }
 
+// GetModelByNameAndOwner returns the model associated with the given model name and owner name.
+// The following errors may be returned:
+// - [modelerrors.NotFound] if no model exists
+func (s *Service) GetModelByNameAndOwner(ctx context.Context, name string, ownerName coreuser.Name) (coremodel.Model, error) {
```

Comment:
You should validate the user name with the tools in that package before passing it down to state. This set of errors will need to be documented as well.

May we please also get some unit tests for this?

Reply:
done thanks tom!